server/leafnode.go

```diff
@@ -1117,6 +1121,13 @@ func (s *Server) initLeafNodeSmap(c *client) int {
 	// if this is coming back to us.
 	c.leaf.smap[lds]++
 
+	// Check if we need to add an existing siReply to our map.
+	// This will be a prefix so add on the wildcard.
+	if siReply != nil {
+		wcsub := append(siReply, '>')
+		c.leaf.smap[string(wcsub)]++
```

Comment:
What about if any server in the super-cluster is restarted (same issue that when we need to know that any node in the super-cluster starts so that we switch GW to interest-only mode for leafnode accounts)?

Reply:
Could take a closer look. Since the code in question that dealt with the siReply on the fly dealt with routes, GWs and LNs.